tests/drivers/adc/adc_accuracy_test/src/ref_volt.c

```diff
@@ -24,6 +24,15 @@ static int test_ref_to_adc(void)
 
 	const struct adc_dt_spec *adc_channel = get_adc_channel();
 
+	if (CONFIG_HAS_HW_NRF_SAADC) {
+		struct adc_sequence calibration_sequence = sequence;
+
+		calibration_sequence.calibrate = true;
+		adc_sequence_init_dt(adc_channel, &calibration_sequence);
+		ret = adc_read_dt(adc_channel, &calibration_sequence);
+		zassert_equal(ret, 0, "adc_read_dt() failed during calibration with code %d", ret);
+	}
```

Comment:
Why not just set `sequence.calibrate` to `true` instead of all this? This flag is supposed to be ignored by drivers that don't support calibration: https://github.com/zephyrproject-rtos/zephyr/blob/6211de87c0660a5dfd21b5decc93a67236ec0984/include/zephyr/drivers/adc.h#L666-L670

Reply:
It seems to me that additional sequence read is needed for calibration, so no all of this can be dropped.
i.e. adc_nrfx_saadc.c triggers different saadc tasks during sampling depending on whether this field is set or not.

So I understand that we need to run the sampling once to calibrate the adc before the actual 'valuable' sampling.

Reply:
No, API specifies that the calibration is to be performed by the driver before the actual ("valuable") sampling is done: https://github.com/zephyrproject-rtos/zephyr/blob/f937e548a7edf60f5eaeac0ce967f55a3325477f/include/zephyr/drivers/adc.h#L663

>  i.e. adc_nrfx_saadc.c triggers different saadc tasks during sampling depending on whether this field is set or not.

`NRF_SAADC_TASK_SAMPLE` is triggered after the calibration [completes](https://github.com/zephyrproject-rtos/zephyr/blob/f937e548a7edf60f5eaeac0ce967f55a3325477f/drivers/adc/adc_nrfx_saadc.c#L627-L639).

Reply:
right, thanks, I missed it